lib/client_summarizer.rb

```diff
@@ -0,0 +1,31 @@
+class ClientSummarizer
+  include FiscalYearMixin
+
+  attr_reader :client_namespace, :fiscal_year
+
+  def initialize(args)
+    @client_namespace = args[:client_namespace]
+    now = Time.zone.now
+    @fiscal_year = (args[:fiscal_year] || self.class.which_fiscal_year(now.year, now.month)).to_i
+    @_records = build_records
+  end
+
+  def run
+    @_summary ||= build_summary
+  end
+
+  private
+
+  def build_records
+    Proposal.for_fiscal_year(fiscal_year).where("client_data_type LIKE ?", "#{client_namespace}::%")
```

Comment:
I might memoize this and just call `build_records` directly in `build_summary` rather than in `initialize` -- what do you think?